File: rust/tcl-registry/src/commands/iapps/script__init.rs
//! `script::init` command.
use crate::prelude::*;
const FORMS: &[FormSpec] = &[FormSpec {
    kind: FormKind::Default,
    synopsis: "script::init",
}];

pub fn spec() -> CommandSpec {
    CommandSpec {
        name: "script::init",
        dialects: Some(DialectSet::IAPPS),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet::brief(
            "Called before ``script::run``, ``script::help``, and ``script::tabc``.",
            &["script::init"],
            "F5",
        )),
        forms: FORMS,
        ..CommandSpec::DEFAULT
    }
}

File: dialects/f5/iapps/script__init.py
"""script::init -- F5 TMSH script namespace command."""

from __future__ import annotations

from compiler.registry._base import CommandDef
from compiler.registry.models import CommandSpec

from ._base import _tmsh_spec, register


@register
class ScriptInit(CommandDef):
    name = "script::init"

    @classmethod
    def spec(cls) -> CommandSpec:
        return _tmsh_spec(
            "script::init",
            "Called before ``script::run``, ``script::help``, and ``script::tabc``.",
            "script::init",
        )
